dandischema/datacite.py

```diff
@@ -134,7 +134,7 @@ def to_datacite(
                 NAME_PATTERN, contr_el.name
             ).pop()
 
-            if getattr(contr_el, "affiliation"):
+            if hasattr(contr_el, "affiliation") and contr_el.affiliation is not None:
```

Comment:
can this not be replaced by the following pattern?

```suggestion
            if getattr(contr_el, "affiliation", None):
```

Reply:
If using `getattr`, mypy can't tell whether `contr_el.affiliation` is `None` or not, so it'll warn when it's used in a non-`None` way inside the block.

Reply:
that sounds like a deficiency in `mypy` since logically it cannot be `None` since then `if` would not evaluate to True.  Do you know of the issue may be already in mypy on this?